### DISCUSSION TITLE
Include minor version or build number

### DISCUSSION THREAD

**Discussion by wyatt-wong** [mamedev/discussions#150] [General]
I checked the latest GitHub source several times per day or daily and I found there are frequently updates to the source. However, after I built the GitHub source to binary in macOS Sequoia, I found it remains version 0.270.

Is it possible to include minor version or build number to differentiate every single change in the GitHub source ?

`
wyattwongMacBookPro mame % git pull            
remote: Enumerating objects: 11, done.
remote: Counting objects: 100% (11/11), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 11 (delta 8), reused 8 (delta 8), pack-reused 0 (from 0)
Unpacking objects: 100% (11/11), 4.29 KiB | 26.00 KiB/s, done.
From https://github.com/mamedev/mame
   8b863228355..1ca12d518a9  master     -> origin/master
Updating 8b863228355..1ca12d518a9
Fast-forward
 hash/famicom_flop.xml    |  2 +-
 hash/nes.xml             | 42 ++++++++-------------
 src/mame/misc/mosaic.cpp | 97 +++++++++++++++++++++---------------------------
 3 files changed, 58 insertions(+), 83 deletions(-)
wyattwongMacBookPro mame %
`

**Comment by cuavas** [comment]
The commit hash is included in the version number unless you specifically tell the build system not to:
```
% ./mamed -version
0.269 (mame0269-327-ga916fd601df-dirty)
```

That build was from the commit mamedev/mame@a916fd601df, which was 327 commits on top of the `mame0269` tag, and there were local source changes not committed.

**Reply by wyatt-wong** [reply]
> The commit hash is included in the version number unless you specifically tell the build system not to:
> 
> ```
> % ./mamed -version
> 0.269 (mame0269-327-ga916fd601df-dirty)
> ```
> 
> That build was from the commit [mamedev/mame@a916fd601df](https://github.com/mamedev/mame/commit/a916fd601df), which was 327 commits on top of the `mame0269` tag, and there were local source changes not committed.

I see that.

**Reply by appleiifanclub** [reply]
> The commit hash is included in the version number unless you specifically tell the build system not to:
> 
> ```
> % ./mamed -version
> 0.269 (mame0269-327-ga916fd601df-dirty)
> ```
> 
> That build was from the commit [mamedev/mame@a916fd601df](https://github.com/mamedev/mame/commit/a916fd601df), which was 327 commits on top of the `mame0269` tag, and there were local source changes not committed.

I didn't find there is a file named "mamed" but "mame'. I am not sure how you can return the version information by executing "./mamed -version"

**Reply by cuavas** [reply]
Builds with asserts enabled have a `d` suffix.  The `-version` option works with builds without asserts enabled as well.
